### ISSUE TITLE
improving readthedocs manual

### ISSUE THREAD

**Issue by mikkokotila** [autonomio/core-module#50]
Read the docs manual (docs/index.rst) could be cleaned up. Also it's better if the menu on the left is structured properly in a way where subheadings are hidden until you are clicking them.

**Comment by mikkokotila** [comment]
@pdey could you look at the [current manual](http://autonomio.readthedocs.io/en/latest/) and give some critical feedback on how to make it (much) better. Because it's something that can be done well, I want to make sure that we present the gold standard in clarity and thoroughness in terms of user docs. One interesting challenge is different levels of users. 

I think it will need a separate issue (or few) but the way I see this is that once the docs are 100% in line with the solution, then the next steps are: 

- code fragments
- notebooks which have code fragments but also commentary
- tutorials both text and video
- zero-doubt code comments
- clear non-python error messages

All feedback to this topic is most appreciated :)

**Comment by mikkokotila** [comment]
This is going to the CONTRIBUTION.md in the next merge, but will post it here for now: 

### 2.3. Documentation standards <a name="docs_for_review"></a>

Because Autonomio is to sombased on creating a very high level abstraction, and a lot of "magic" takes place without the user knowing about it, it's very important that we are elaborate in our documentation. This needs to take place to the extent that users like the one who made the following comment are happy: 

> Strong documentation and tutorials, with an emphasis on data prep. Don't abstract things at the cost of understanding what is going on behind the hood. I want to know how it all works, and what is being done to my data - I just don't want to have to code every step.

This is the kind of situation we are trying to create; a lot of mundane but hard or time consuming to do things are taken care of on behalf of the user, but if the user so desires, he or she can get a full (and clear!) picture of what's being done. 

In order to pass code review (required for merging pull requests), together with your code in the same pull request you must provide the following updates to the documentation [docs/index.rst](https://github.com/mikkokotila/core-module/blob/master/docs/index.rst). 

1) What does it do (a high level overview) 
  1.1) some use examples
2) What data input is requires
  2.1) some use examples 
3) What it outputs 
4) What parameters are there (a high level overview)
  4.1) What options each parameter has
  4.2) What is the function of each parameter option

In short summary, the goal is that the user can completely understand 100% of the functioning of the features. If something is done so that the user can’t see it when they are running the commands, we have to explain exactly and thoroughly what kind of automations are taking place.

**Comment by mikkokotila** [comment]
New docs are [here](https://github.com/autonomio/core-module)
